### PR TITLE
Separate  Ide Classes

### PR DESCRIPTION
**Pull request by fahad-israr**
- Make separate classes for each IDE

### CODE REVIEW

File: cli/src/main/java/dev/starfix/Starfix.java [hunk 1]
```diff
@@ -288,37 +290,108 @@ public void setClone_path(String clone_path) {
...
+        GenericIDE(String ide){
+            setIDE(ide);
+        }
+
+        public IDE getIDE(){
+            //Return Current IDE
+            return ide;
+        }
+
+        public IDE getIDE(String ide){
```
Comment (maxandersen) [inline cli/src/main/java/dev/starfix/Starfix.java:315]: seems there is nothing in GenericIDE besides just this method - would that not be sufficient to just have getIDE method and it returns the specific subclass as needed ?

Comment (fahad-israr) [reply cli/src/main/java/dev/starfix/Starfix.java:315]: I actually made a method but then came across usage  of a generic launcher. Like other IDEs(VsCode, Idea etc) this class also has its launcher. You'd notice I am using that launcher in other IDE classes as well.

Comment (maxandersen) [reply cli/src/main/java/dev/starfix/Starfix.java:315]: But we just need one instance; why not just set a ide variable on the command and use that directly ?

Comment (fahad-israr) [reply cli/src/main/java/dev/starfix/Starfix.java:315]: Ok Max, I'll replace that GenericIDE portion with a getIDE method.

**Comment by fahad-israr** [comment]
I've made the suggested changes .

**Review by maxandersen — APPROVED** [review]

File: cli/src/main/java/dev/starfix/Starfix.java [hunk 2]
```diff
@@ -101,6 +101,8 @@ public int cloneCmd(@Parameters(index = "0") String url) {
             filePath = Paths.get(clone_path,repo_name,filePath).toAbsolutePath().toString();
 
             // Launching Editor on the Cloned Directory
+            System.out.println("Launching  Editor Now...");
+            getIDE(ide).launch_editor(directory, ide, directory.toAbsolutePath().toString(),filePath);
             System.out.println("Opening " + filePath);
```
Comment (maxandersen) [inline cli/src/main/java/dev/starfix/Starfix.java:106]:
```suggestion
```

File: cli/src/main/java/dev/starfix/Starfix.java [hunk 3]
```diff
@@ -101,6 +101,8 @@ public int cloneCmd(@Parameters(index = "0") String url) {
...
             launch_editor(directory, ide, directory.toAbsolutePath().toString(),filePath);
```
Comment (maxandersen) [inline cli/src/main/java/dev/starfix/Starfix.java:107]:
```suggestion
```

**Pull request merged**